src/EditorFeatures/Core/Diagnostics/DiagnosticsClassificationTaggerProvider.cs

```diff
@@ -31,15 +31,14 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     [ContentType(ContentTypeNames.RoslynContentType)]
     [ContentType(ContentTypeNames.XamlContentType)]
     [TagType(typeof(ClassificationTag))]
-    internal partial class DiagnosticsClassificationTaggerProvider : AbstractDiagnosticsTaggerProvider<ClassificationTag>
+    internal sealed partial class DiagnosticsClassificationTaggerProvider : AbstractDiagnosticsTaggerProvider<ClassificationTag>
     {
-        private static readonly IEnumerable<Option2<bool>> s_tagSourceOptions = ImmutableArray.Create(EditorComponentOnOffOptions.Tagger, InternalFeatureOnOffOptions.Classification);
-
         private readonly ClassificationTypeMap _typeMap;
         private readonly ClassificationTag _classificationTag;
         private readonly EditorOptionsService _editorOptionsService;
 
-        protected override IEnumerable<Option2<bool>> Options => s_tagSourceOptions;
+        protected override ImmutableArray<IOption> Options { get; } =
+            ImmutableArray.Create<IOption>(EditorComponentOnOffOptions.Tagger, InternalFeatureOnOffOptions.Classification);
```

Comment:
I think `EditorComponentOnOffOptions.Tagger` is already checked here:

https://github.com/dotnet/roslyn/blob/b9a44fd1d145da55e1b8d8eed6e3be5b24798498/src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.cs#L123-L124

So if I understand correctly, either the above check is redundant, or the existence of `Tagger` option here and in other places is redundant.

If my understanding is correct, then I think there are two options:

1. Remove `EditorComponentOnOffOptions.Tagger` from `Options` and let the condition linked above handle things. This in turn will allow simplifying `Options` from being an immutable array to just a nullable option.
2. Remove the redundant condition linked above, the downside of that is that new types that extend (directly or indirectly) from `AbstractAsynchronousTaggerProvider` will have to remember including the `EditorComponentOnOffOptions.Tagger`

Reply:
I went with '1'.  I'm not going to simplify down to a nullable option as i don't want to have to go through teh exercise of making this into an array if we have something that checks multiple optinos.